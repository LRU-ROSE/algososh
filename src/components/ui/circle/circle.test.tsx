import renderer from "react-test-renderer";

import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Тестирование компонента Circle", () => {
  it("отрисовка без буквы", () => {
    const tree = renderer.create(<Circle letter="" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка с буквами", () => {
    const tree = renderer.create(<Circle letter="1" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка с head", () => {
    const tree = renderer.create(<Circle letter="H" head="head" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка с react-элементом в head", () => {
    const tree = renderer
      .create(<Circle letter="P" head={<Circle letter="C" />} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка с tail", () => {
    const tree = renderer.create(<Circle letter="T" tail="tail" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка с react-элементом в tail", () => {
    const tree = renderer
      .create(<Circle letter="T" tail={<Circle letter="t" />} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка с index", () => {
    const tree = renderer.create(<Circle index={1} letter="a" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка с пропом isSmall === true", () => {
    const tree = renderer.create(<Circle isSmall={true} letter="s" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка в состоянии default", () => {
    const tree = renderer
      .create(<Circle letter="D" state={ElementStates.Default} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка в состоянии changing", () => {
    const tree = renderer
      .create(<Circle letter="C" state={ElementStates.Changing} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("отрисовка в состоянии modified", () => {
    const tree = renderer
      .create(<Circle letter="M" state={ElementStates.Modified} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
