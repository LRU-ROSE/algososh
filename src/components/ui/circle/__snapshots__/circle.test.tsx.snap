// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle отрисовка без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-type="tail"
  />
</div>
`;

exports[`Тестирование компонента Circle отрисовка в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      C
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-type="tail"
  />
</div>
`;

exports[`Тестирование компонента Circle отрисовка в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      D
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-type="tail"
  />
</div>
`;

exports[`Тестирование компонента Circle отрисовка в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      M
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-type="tail"
  />
</div>
`;

exports[`Тестирование компонента Circle отрисовка с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-type="head"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      H
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-type="tail"
  />
</div>
`;

exports[`Тестирование компонента Circle отрисовка с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      a
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-type="tail"
  />
</div>
`;

exports[`Тестирование компонента Circle отрисовка с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-type="head"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-type="letter"
        >
          C
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-type="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      P
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-type="tail"
  />
</div>
`;

exports[`Тестирование компонента Circle отрисовка с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      T
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-type="tail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-type="head"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-type="letter"
        >
          t
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-type="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle отрисовка с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      T
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-type="tail"
  >
    tail
  </div>
</div>
`;

exports[`Тестирование компонента Circle отрисовка с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      1
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-type="tail"
  />
</div>
`;

exports[`Тестирование компонента Circle отрисовка с пропом isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-type="head"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-type="letter"
    >
      s
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-type="tail"
  />
</div>
`;
